src/klu.jl

```diff
@@ -0,0 +1,555 @@
+                # reset the last entry
+                Fp[end] += 1
+            end
+        end
+        return Fp, Fi, Fx, Fz
+    end
+    if s ∈ [:q, :p, :R, :Rs]
+        if s === :Rs
+            out = Vector{Float64}(undef, klu.n)
+        elseif s === :R
+            out = Vector{Ti}(undef, klu.nblocks + 1)
+        else
+            out = Vector{Ti}(undef, klu.n)
+        end
+        # This tuple construction feels hacky, there's a better way I'm sure.
+        s === :q && (s = :Q)
+        s === :p && (s = :P)
+        _extract!(klu; NamedTuple{(s,)}((out,))...)
+        if s ∈ [:Q, :P, :R]
+            increment!(out)
+        end
+        return out
+    end
+    if s ∈ [:L, :U, :F]
+        if s === :L
+            p, i, x, z = klu._L
+        elseif s === :U
+            p, i, x, z = klu._U
+        elseif s === :F
+            p, i, x, z = klu._F
+        end
+        if Tv == Float64
+            return SparseMatrixCSC(klu.n, klu.n, increment!(p), increment!(i), x)
+        else
+            return SparseMatrixCSC(klu.n, klu.n, increment!(p), increment!(i), Complex.(x, z))
+        end
+    end
+end
+
+function LinearAlgebra.issuccess(K::KLUFactorization)
+    return K.common.status == KLU_OK && K._numeric != C_NULL
+end
+function show(io::IO, mime::MIME{Symbol("text/plain")}, K::KLUFactorization)
+    if issuccess(K)
+        summary(io, K); println(io)
+        println(io, "L factor:")
+        show(io, mime, K.L)
+        println(io, "\nU factor:")
+        show(io, mime, K.U)
+        F = K.F
+        if F !== nothing
+            println(io, "\nF factor:")
+            show(io, mime, K.F)
+        end
+    else
+        print(io, "Failed factorization of type $(typeof(K)). Try `klu_factor!(K)`.")
+    end
+end
+
+function klu_analyze!(K::KLUFactorization{Tv, Ti}) where {Tv, Ti<:KLUITypes}
+    if K._symbolic != C_NULL return K end
+    if Ti == Int64
+        sym = klu_l_analyze(K.n, K.colptr, K.rowval, Ref(K.common))
+    else
+        sym = klu_analyze(K.n, K.colptr, K.rowval, Ref(K.common))
+    end
+    if sym == C_NULL
+        kluerror(K.common)
+    else
+        K._symbolic = sym
+    end
+    return K
+end
+
+for Tv ∈ KLUValueTypes, Ti ∈ KLUIndexTypes
+    factor = _klu_name("factor", Tv, Ti)
+    @eval begin
+        function klu_factor!(K::KLUFactorization{$Tv, $Ti})
+            K._symbolic == C_NULL  && klu_analyze!(K)
+            num = $factor(K.colptr, K.rowval, K.nzval, K._symbolic, Ref(K.common))
+            if num == C_NULL
+                kluerror(K.common)
+            else
+                K._numeric = num
+            end
+            return K
+        end
+    end
+end
+
+"""
+klu_factor!(K::KLUFactorization)
+
+Factor `K` into components `K.L`, `K.U`, and `K.F`.
+This function will perform both the symbolic and numeric steps of factoriation on an
+    existing `KLUFactorization` instance.
+
+    The `K.common` struct can be used to modify certain options and parameters such as:
+
```

Comment:
```suggestion
existing `KLUFactorization` instance.

The `K.common` struct can be used to modify certain options and parameters such as:

```
Alos, this looks like unfinished.